docs/sender.rst

```diff
@@ -214,172 +282,227 @@ The transaction is automatically committed when the ``with`` block is exited.
-  used in production as it makes the connection vulnerable to man-in-the-middle
-  attacks.
-  
-  The default is: ``'on'``.
+    conf = 'http::addr=localhost:9000;'
+    with Sender.from_conf(conf) as sender:
+        sender.flush(buf, transaction=True)
 
-.. _configuring-sender-auto-flushing:
+The ``transaction`` parameter is optional and defaults to ``False``.
+When set to ``True``, the buffer is guaranteed to be committed as a single
+transaction, but must only contain rows for a single table.
 
-Auto-flushing
--------------
+Multiple Databases
+------------------
 
-The following paramers control the :ref:`sender-auto-flushing` behavior.
+Handling buffers explicitly is also useful when sending data to multiple
+databases via the `.flush(buf, clear=False)` option.
 
-* ``auto_flush`` - ``'on'`` | ``'off'``: Global switch for the auto-flushing
-  behavior.
+.. code-block:: python
 
-  *Default: ``'on'``.*
+    from questdb.ingress import Buffer, Sender, TimestampNanos
 
-* ``auto_flush_rows`` - ``int > 0`` | ``'off'``: The number of rows that will
-  trigger a flush. Set to ``'off'`` to disable.
-    
-  *Default: 75000 (HTTP) | 600 (TCP).*
+    buf = Buffer()
+    buf.row(
+        'weather_sensor',
+        symbols={'id': 'toronto1'},
+        columns={'temperature': 23.5, 'humidity': 0.49},
+        at=TimestampNanos.now())
 
-* ``auto_flush_bytes`` - ``int > 0`` | ``'off'``: The number of bytes that will
-  trigger a flush. Set to ``'off'`` to disable.
-        
-  *Default: ``'off'``.*
+    conf1 = 'http::addr=db1.host.com:9000;'
+    conf2 = 'http::addr=db2.host.com:9000;'
+    with Sender.from_conf(conf1) as sender1, Sender.from_conf(conf2) as sender2:
+        sender1.flush(buf1, clear=False)
+        sender2.flush(buf2, clear=False)
 
-* ``auto_flush_interval`` - ``int > 0`` | ``'off'``: The time in milliseconds
-  that will trigger a flush. Set to ``'off'`` to disable.
-    
-  *Default: 1000 (millis).*
+    buf.clear()
 
-.. _configuring-sender-buffer:
+This uses the ``clear=False`` parameter which otherwise defaults to ``True``.
 
-Buffer
-------
+Threading Considerations
+------------------------
 
-* ``init_buf_size`` - ``int > 0``: Initial buffer capacity.
-    
-  *Default: 65536 (64KiB).*
+Neither buffer API nor the sender object are thread-safe, but can be shared
+between threads if you take care of exclusive access (such as using a lock)
+yourself.
 
-* ``max_buf_size`` - ``int > 0``: Maximum flushable buffer capacity.
-    
-  *Default: 104857600 (100MiB).*
+Independent buffers also allows you to prepare separate buffers in different
+threads and then send them later through a single exclusively locked sender.
 
-* ``max_name_len`` - ``int > 0``: Maximum length of a table or column name.
+Alternatively you can also create multiple senders, one per thread.
 
-  *Default: 127.*
+Notice that the ``questdb`` python module is mostly implemented in Cython and
```

Comment:
This context is neat. And it is high complexity.

It also does not appear user-actionable.

Share between threads vs. create multiple senders, one-per-thread.

Declaring options, then demonstrating, would be most helpful.

Reply:
I've dropped Cython and Rust and simply stated "native code".

This is important to a Python developer since it gives them guidance on the coding style
that is possible to extract best performance.